extension/observer/cfgardenobserver/extension.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package cfgardenobserver // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer/cfgardenobserver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+type cfGardenObserver struct {
+	*observer.EndpointsWatcher
+}
+
+var _ extension.Extension = (*cfGardenObserver)(nil)
+
+func newObserver(params extension.Settings, config *Config) (extension.Extension, error) {
+	g := &cfGardenObserver{}
+	g.EndpointsWatcher = observer.NewEndpointsWatcher(g, time.Second, params.Logger)
+
+	return g, nil
+}
+
+func (g *cfGardenObserver) Start(context.Context, component.Host) error {
```

Comment:
If you want to make it easier on yourself if you don't need to implement `Start` and `Shutdown`, you can embed `component.StartFunc` and `component.ShutdownFunc` within the main type and it will implement safe methods for external methods to call.